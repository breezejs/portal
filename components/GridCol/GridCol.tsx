import React, {memo, ReactNode} from 'react';
import styled from 'styled-components';

const Col = styled.div`
  grid-column: ${({xs}: IGridColProps) => `span ${xs}`};
  position: relative;

  ${({sm}: IGridColProps) => sm ? `
    @media (min-width: 576px) {
      grid-column: span ${sm};
    }
  ` : null}

  ${({md}: IGridColProps) => md ? `
    @media (min-width: 768px) {
      grid-column: span ${md};
    }
  ` : null}

  ${({lg}: IGridColProps) => lg ? `
    @media (min-width: 992px) {
      grid-column: span ${lg};
    }
  ` : null}
`;

type GridColType =
  | 1
  | 2
  | 3
  | 4
  | 5
  | 6
  | 7
  | 8
  | 9
  | 10
  | 11
  | 12;

interface IGridColProps {
  children?: ReactNode;
  lg?: GridColType;
  md?: GridColType;
  sm?: GridColType;
  xs?: GridColType;
}

function GridCol ({children, lg, md, sm, xs = 12}: IGridColProps) {
  return (
    <Col lg={lg} md={md} sm={sm} xs={xs}>{children}</Col>
  );
}

export default memo(GridCol);
