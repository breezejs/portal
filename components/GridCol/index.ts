export {default as default} from './GridCol';
