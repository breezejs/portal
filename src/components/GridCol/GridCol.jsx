import PropTypes from 'prop-types';
import React, {memo} from 'react';
import styled from 'styled-components';

const Col = styled.div`
  grid-column: ${({xs}) => `span ${xs}`};
  position: relative;

  ${({sm}) => sm ? `
    @media (min-width: 576px) {
      grid-column: span ${sm};
    }
  ` : null}

  ${({md}) => md ? `
    @media (min-width: 768px) {
      grid-column: span ${md};
    }
  ` : null}

  ${({lg}) => lg ? `
    @media (min-width: 992px) {
      grid-column: span ${lg};
    }
  ` : null}
`;

function GridCol ({children, lg, md, sm, xs}) {
  return (
    <Col lg={lg} md={md} sm={sm} xs={xs}>{children}</Col>
  );
}

GridCol.propTypes = {
  children: PropTypes.node,
  lg: PropTypes.number,
  md: PropTypes.number,
  sm: PropTypes.number,
  xs: PropTypes.number
};

GridCol.defaultProps = {
  children: null,
  lg: null,
  md: null,
  sm: null,
  xs: 12
};

export default memo(GridCol);
