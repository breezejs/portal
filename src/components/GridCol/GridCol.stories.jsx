import {number, withKnobs} from '@storybook/addon-knobs';
import {storiesOf} from '@storybook/react';
import React from 'react';
import GlobalStyles from '../GlobalStyles';
import Grid from '../Grid';
import GridCol from './GridCol';

const stories = storiesOf('GridCol', module);

stories
  .addDecorator(withKnobs)
  .addDecorator((storyFn) => (
    <div>
      <GlobalStyles />
      {storyFn()}
    </div>
  ))
  .add('with responsive widths', () => (
    <Grid>
      <GridCol xs={number('xs', 12)} sm={number('sm', 10)} md={number('md', 8)} lg={number('lg', 6)}>Column</GridCol>
    </Grid>
  ));
