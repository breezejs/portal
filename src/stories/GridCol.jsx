import {number, withKnobs} from '@storybook/addon-knobs';
import {storiesOf} from '@storybook/react';
import React from 'react';
import GlobalStyles from '../components/GlobalStyles';
import Grid from '../components/Grid';
import GridCol from '../components/GridCol';

const stories = storiesOf('GridCol', module);

stories
  .addDecorator(withKnobs)
  .addDecorator((storyFn) => (
    <div>
      <GlobalStyles />
      {storyFn()}
    </div>
  ))
  .add('with responsive widths', () => (
    <Grid>
      <GridCol xs={number('xs', 12)} sm={number('sm', 10)} md={number('md', 8)} lg={number('lg', 6)}>Column</GridCol>
    </Grid>
  ));
