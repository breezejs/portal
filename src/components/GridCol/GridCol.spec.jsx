import {mount, shallow} from 'enzyme';
import React from 'react';
import GridCol from './GridCol';

describe('GridCol', () => {
  describe('with default values', () => {
    it('should render', () => {
      const result = shallow(
        <GridCol />
      );

      expect(result).toMatchSnapshot();
    });

    it('should render with children', () => {
      const result = mount(
        <GridCol>Hello world</GridCol>
      );

      expect(result).toMatchSnapshot();
    });

    it('should have the correct css styles', () => {
      const result = shallow(
        <GridCol />
      );

      expect(result.find('div')).toMatchSnapshot();
    });
  });

  describe('with xs set', () => {
    it('should render', () => {
      const result = shallow(
        <GridCol xs={6} />
      );

      expect(result).toMatchSnapshot();
    });

    it('should render with children', () => {
      const result = mount(
        <GridCol xs={6}>Hello world</GridCol>
      );

      expect(result).toMatchSnapshot();
    });

    it('should have the correct css styles', () => {
      const result = shallow(
        <GridCol xs={6} />
      );

      expect(result.find('div')).toMatchSnapshot();
    });
  });

  describe('with sm set', () => {
    it('should render', () => {
      const result = shallow(
        <GridCol sm={6} />
      );

      expect(result).toMatchSnapshot();
    });

    it('should render with children', () => {
      const result = mount(
        <GridCol sm={6}>Hello world</GridCol>
      );

      expect(result).toMatchSnapshot();
    });

    it('should have the correct css styles', () => {
      const result = shallow(
        <GridCol sm={6} />
      );

      expect(result.find('div')).toMatchSnapshot();
    });
  });

  describe('with md set', () => {
    it('should render', () => {
      const result = shallow(
        <GridCol sm={6} md={4} />
      );

      expect(result).toMatchSnapshot();
    });

    it('should render with children', () => {
      const result = mount(
        <GridCol sm={6} md={4}>Hello world</GridCol>
      );

      expect(result).toMatchSnapshot();
    });

    it('should have the correct css styles', () => {
      const result = shallow(
        <GridCol sm={6} md={4} />
      );

      expect(result.find('div')).toMatchSnapshot();
    });
  });

  describe('with lg set', () => {
    it('should render', () => {
      const result = shallow(
        <GridCol sm={6} md={4} lg={2} />
      );

      expect(result).toMatchSnapshot();
    });

    it('should render with children', () => {
      const result = mount(
        <GridCol sm={6} md={4} lg={2}>Hello world</GridCol>
      );

      expect(result).toMatchSnapshot();
    });

    it('should have the correct css styles', () => {
      const result = shallow(
        <GridCol sm={6} md={4} lg={2} />
      );

      expect(result.find('div')).toMatchSnapshot();
    });
  });
});
