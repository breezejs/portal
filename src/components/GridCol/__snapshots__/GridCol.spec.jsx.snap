// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridCol with default values should have the correct css styles 1`] = `null`;

exports[`GridCol with default values should render 1`] = `
<styled.div
  xs={12}
/>
`;

exports[`GridCol with default values should render with children 1`] = `
.c0 {
  grid-column: span 12;
  position: relative;
}

<GridCol>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  grid-column: ",
            [Function],
            ";
  position: relative;

  ",
            [Function],
            "

  ",
            [Function],
            "

  ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    xs={12}
  >
    <div
      className="c0"
    >
      Hello world
    </div>
  </StyledComponent>
</GridCol>
`;

exports[`GridCol with lg set should have the correct css styles 1`] = `null`;

exports[`GridCol with lg set should render 1`] = `
<styled.div
  lg={2}
  md={4}
  sm={6}
  xs={12}
/>
`;

exports[`GridCol with lg set should render with children 1`] = `
.c0 {
  grid-column: span 12;
  position: relative;
}

@media (min-width:576px) {
  .c0 {
    grid-column: span 6;
  }
}

@media (min-width:768px) {
  .c0 {
    grid-column: span 4;
  }
}

@media (min-width:992px) {
  .c0 {
    grid-column: span 2;
  }
}

<GridCol
  lg={2}
  md={4}
  sm={6}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  grid-column: ",
            [Function],
            ";
  position: relative;

  ",
            [Function],
            "

  ",
            [Function],
            "

  ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    lg={2}
    md={4}
    sm={6}
    xs={12}
  >
    <div
      className="c0"
    >
      Hello world
    </div>
  </StyledComponent>
</GridCol>
`;

exports[`GridCol with md set should have the correct css styles 1`] = `null`;

exports[`GridCol with md set should render 1`] = `
<styled.div
  md={4}
  sm={6}
  xs={12}
/>
`;

exports[`GridCol with md set should render with children 1`] = `
.c0 {
  grid-column: span 12;
  position: relative;
}

@media (min-width:576px) {
  .c0 {
    grid-column: span 6;
  }
}

@media (min-width:768px) {
  .c0 {
    grid-column: span 4;
  }
}

<GridCol
  md={4}
  sm={6}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  grid-column: ",
            [Function],
            ";
  position: relative;

  ",
            [Function],
            "

  ",
            [Function],
            "

  ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    md={4}
    sm={6}
    xs={12}
  >
    <div
      className="c0"
    >
      Hello world
    </div>
  </StyledComponent>
</GridCol>
`;

exports[`GridCol with sm set should have the correct css styles 1`] = `null`;

exports[`GridCol with sm set should render 1`] = `
<styled.div
  sm={6}
  xs={12}
/>
`;

exports[`GridCol with sm set should render with children 1`] = `
.c0 {
  grid-column: span 12;
  position: relative;
}

@media (min-width:576px) {
  .c0 {
    grid-column: span 6;
  }
}

<GridCol
  sm={6}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  grid-column: ",
            [Function],
            ";
  position: relative;

  ",
            [Function],
            "

  ",
            [Function],
            "

  ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    sm={6}
    xs={12}
  >
    <div
      className="c0"
    >
      Hello world
    </div>
  </StyledComponent>
</GridCol>
`;

exports[`GridCol with xs set should have the correct css styles 1`] = `null`;

exports[`GridCol with xs set should render 1`] = `
<styled.div
  xs={6}
/>
`;

exports[`GridCol with xs set should render with children 1`] = `
.c0 {
  grid-column: span 6;
  position: relative;
}

<GridCol
  xs={6}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  grid-column: ",
            [Function],
            ";
  position: relative;

  ",
            [Function],
            "

  ",
            [Function],
            "

  ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    xs={6}
  >
    <div
      className="c0"
    >
      Hello world
    </div>
  </StyledComponent>
</GridCol>
`;
